File: hto/61/hto_61.java
class Solution {
    public boolean isStraight(int[] nums) {
        Set<Integer> repeat = new HashSet<>();
        int max = 0, min = 14;
        for(int num : nums) {
            if(num == 0) continue; 
            max = Math.max(max, num); 
            min = Math.min(min, num); 
            if(repeat.contains(num)) return false;
            repeat.add(num); 
        }
        return max - min < 5; 
    }
}

File: hto/61/hto_61.py
from typing import List, Dict
from queue import Queue
from collections import deque, Counter
import math
from heapq import heappush, heappushpop

class Solution:
    def isStraight(self, nums: List[int]) -> bool:
        repeat = set()
        ma, mi = 0, 14
        for num in nums:
            if num == 0:
                continue
            ma = max(ma, num) 
            mi = min(mi, num)
            if num in repeat:
                return False 
            repeat.add(num) 
        return ma - mi < 5

